install.ps1

```diff
@@ -1,88 +1,66 @@
+#!/usr/bin/env pwsh
 # Copyright 2018 the Deno authors. All rights reserved. MIT license.
 # TODO(everyone): Keep this script simple and easily auditable.
+
 param (
-  [alias("v")]
-  [string] $version
+  [ValidatePattern('^v(\d+).(\d+).(\d+)$')]
+  [String] $Version = 'v0.2.6'
 )
 
-$ErrorActionPreference = "Stop"
-
-# Enable TLS 1.2 since it is required for connections to GitHub.
-[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$ErrorActionPreference = 'Stop'
 
-# Helper functions for pretty terminal output.
-function Write-Part ([string] $Text) {
-  Write-Host $Text -NoNewline
-}
-function Write-Emphasized ([string] $Text) {
-  Write-Host $Text -NoNewLine -ForegroundColor "Yellow"
-}
-function Write-Done {
-  Write-Host " done" -NoNewline -ForegroundColor "Green";
-  Write-Host "."
+if ($PSVersionTable.PSVersion.Major -lt 6) {
+  $IsWindows = $true
+  $IsMacOS = $false
 }
 
-if (-not $version) {
-  # Determine latest Deno release via GitHub API.
-  $latest_release_uri = "https://api.github.com/repos/denoland/deno/releases/latest"
-  Write-Part "Downloading "; Write-Emphasized $latest_release_uri; Write-Part "..."
-  $latest_release_json = Invoke-WebRequest -Uri $latest_release_uri
-  Write-Done
+$DenoDir = if ($IsWindows)
+  { "$Home\.deno\bin" } else
+  { "$Home/.deno/bin" }
 
-  Write-Part "Determining latest Deno release: "
-  $version = ($latest_release_json | ConvertFrom-Json).tag_name
-  Write-Emphasized $version; Write-Part "... "
-  Write-Done
-}
+$Zip = if ($IsWindows)
+  { 'zip' } else
+  { 'gz' }
 
-# Create ~\.deno\bin directory if it doesn't already exist
-$deno_dir = "${Home}\.deno\bin"
-if (-not (Test-Path $deno_dir)) {
-  Write-Part "Creating directory "; Write-Emphasized $deno_dir; Write-Part "..."
-  New-Item -Path $deno_dir -ItemType Directory | Out-Null
-  Write-Done
-}
-
-# Download Deno release.
-$zip_file = "${deno_dir}\deno_win_x64.zip"
-$download_uri = "https://github.com/denoland/deno/releases/download/" +
-                "${version}/deno_win_x64.zip"
-Write-Part "Downloading "; Write-Emphasized $download_uri; Write-Part "..."
-Invoke-WebRequest -Uri $download_uri -OutFile $zip_file
-Write-Done
+$DenoZip = if ($IsWindows)
+  { "$DenoDir\deno.$Zip" } else
```

Comment:
I think a forward slash works fine on Windows too.

Reply:
I'm not sure. Have you tested this with both PowerShell and PowerShell Core on Windows? If so, I'd gladly change it. :)

Unfortunately, I only will have access to my Windows machine on Saturday again.

Reply:
Maybe we should keep the backslashes on Windows since the path of the executable will be printed?